worker/src/test/java/gov/cms/ab2d/worker/processor/JobPreProcessorIntegrationTest.java

```diff
@@ -0,0 +1,119 @@
+import gov.cms.ab2d.common.repository.JobRepository;
+import gov.cms.ab2d.common.repository.SponsorRepository;
+import gov.cms.ab2d.common.repository.UserRepository;
+import gov.cms.ab2d.common.util.AB2DPostgresqlContainer;
+import gov.cms.ab2d.worker.processor.JobPreProcessor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.time.OffsetDateTime;
+import java.util.Random;
+
+import static java.lang.Boolean.TRUE;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest
+@Testcontainers
+class JobPreProcessorIntegrationTest {
+    private Random random = new Random();
+
+    @Autowired
+    private JobPreProcessor cut;
+
+    @Autowired
+    private JobRepository jobRepository;
+    @Autowired
+    private SponsorRepository sponsorRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    private Job job;
+
+    @Container
+    private static final PostgreSQLContainer postgreSQLContainer= new AB2DPostgresqlContainer();
+
+    @BeforeEach
+    void setUp() {
+        jobRepository.deleteAll();
+        userRepository.deleteAll();
+        sponsorRepository.deleteAll();
+
+        var sponsor = createSponsor();
+        var user = createUser(sponsor);
+        job = createJob(user);
+    }
+
+
+    @Test
+    @DisplayName("When a job is in submitted status, it can be put into progress upon starting processing")
+    void whenJobIsInSubmittedStatus_ThenJobShouldBePutInProgress() {
+        var processedJob = cut.preprocess("S001");
+        assertThat(processedJob.getStatus(), is(JobStatus.IN_PROGRESS));
+    }
+
+    @Test
+    @DisplayName("Throw an exception if the job does not exist")
+    void putNonExistentJobInProgress() {
+        var exceptionThrown = assertThrows(
+                IllegalArgumentException.class,
+                () -> cut.preprocess("NonExistent"));
+        assertThat(exceptionThrown.getMessage(), is("Job NonExistent was not found"));
+    }
+
+    @Test
+    @DisplayName("When a job is not already in a submitted status, it cannot be put into progress")
+    void whenJobIsNotInSubmittedStatus_ThenJobShouldNotBePutInProgress() {
+        job.setStatus(JobStatus.IN_PROGRESS);
+        jobRepository.save(job);
+
+        var exceptionThrown = assertThrows(
+                IllegalArgumentException.class,
+                () -> cut.preprocess("S001"));
+
+        assertThat(exceptionThrown.getMessage(), is("Job S001 is not in SUBMITTED status"));
+    }
+
+
+    private Sponsor createSponsor() {
```

Comment:
Just an FYI there's already code in the common module in TestUtil to do some of this stuff. You can inject it into classes like this.

Reply:
I considered using the object-mother with factory methods for creating the test data, but in the end, I prefer to set up my test data in the test class so that it can be specific to the test case instead of using a generic test data. Also, I feel it is a bit odd to create a spring bean just for creating test data. In addition to being unexpected, it also has the potential to become a god object over time if one keeps injecting all the repositories into it. Am going to leave this as is for now, but am happy to revisit this at a later date, if needed.